From 416c7e9facf905c5be757dd1a1c66e2ed878f641 Mon Sep 17 00:00:00 2001
From: "Tobias C. Berner" <tcberner@gmail.com>
Date: Mon, 8 May 2017 15:30:18 +0200
Subject: [PATCH] Change default pkgconfig install path for FreeBSD

Summary:
FreeBSD, opposed to most other sytems, does not install pkgconfig files to
lib/pkgconfig, but to libdata/pkgconfig.

Reviewers: #freebsd, apol, kfunk

Reviewed By: kfunk

Subscribers: kfunk, #frameworks, #build_system

Tags: #frameworks, #build_system

Differential Revision: https://phabricator.kde.org/D5766
---
 modules/ECMGeneratePkgConfigFile.cmake | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules/ECMGeneratePkgConfigFile.cmake b/modules/ECMGeneratePkgConfigFile.cmake
index 66c57cd..11f043d 100644
--- modules/ECMGeneratePkgConfigFile.cmake
+++ modules/ECMGeneratePkgConfigFile.cmake
@@ -169,7 +169,11 @@ Requires: ${PKGCONFIG_TARGET_DEPS}
   )
 
   if(EGPF_INSTALL)
-    set(ECM_PKGCONFIG_INSTALL_DIR "${EGPF_LIB_INSTALL_DIR}/pkgconfig" CACHE PATH "The directory where pkgconfig will be installed to.")
+    if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+      set(ECM_PKGCONFIG_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/libdata/pkgconfig" CACHE PATH "The directory where pkgconfig will be installed to.")
+    else()
+      set(ECM_PKGCONFIG_INSTALL_DIR "${EGPF_LIB_INSTALL_DIR}/pkgconfig" CACHE PATH "The directory where pkgconfig will be installed to.")
+    endif()
     install(FILES ${PKGCONFIG_FILENAME} DESTINATION ${ECM_PKGCONFIG_INSTALL_DIR})
   endif()
 endfunction()
-- 
2.12.1

